### DISCUSSION TITLE
WARNING: All projects using nuxt 2 must upgrade to nuxt 3 before September 11th, 2023

### DISCUSSION THREAD

**Discussion by duc2019** [nuxt/nuxt#19078] [General]
This is such a HUGE pain.

Given:

- NuxtJS 2 does NOT support webpack 5: https://github.com/nuxt/nuxt/issues/9268
- Webpack 4 using md4 for hashing filename, and it's hardcode: https://github.com/webpack/webpack/issues/16731
- Using crypto md4 on nodejs 17 will cause error: https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported
- Since nodejs 18 "--openssl-legacy-provider" is no longer supported
- Nodejs 16 will EOL at September 11th, 2023: https://nodejs.org/en/blog/announcements/nodejs16-eol

We have to upgrade nuxt 3 before nodejs 16 EOL, which require us rewrite almost everything.
Why we have to rewrite frontend projects every 2 to 3 years???
In the old days when we using jquery we can have a project supported more than 10 years, this modern javascript world sucks.

**Comment by pi0** [comment]
Indeed it is a huge pain to maintain legacy stuff. Hopefully from Nuxt 3 onwards, we have much better major version upgrades since the core is rewritten to allow this.

Regarding Nuxt 2, don't worry we might finally end up forking webpack 2 for a while if necessary until Nuxt 2 EOL end of this year.

**Reply by duc2019** [reply]
Webpack developer said they won't fix this issue becauase webpack 4 is dead.
I think it's not nuxt.js's fault, more like the javascript community issue these people never consider backward compatibility.
Anyway thanks for your effort, this project is great, I hope it will be much easier to upgrade when nuxtjs 4 is out. (for now it's impossible to inplace upgrade from nuxtjs 2 to nuxtjs 3).

**Comment by danielroe** [comment]
This should be resolved in the [edge channel](https://nuxt.com/docs/guide/going-further/edge-channel/) via https://github.com/nuxt/nuxt/pull/19108, which you can test today with `nuxt-edge`. It will be included in the next release.

**Comment by K-Ramkrupakaran** [comment]
Does Nuxt 2 has any extended support with third party vendor something similar to Vue 2 (https://www.herodevs.com/support/nes-vue) ?

Can someone please confirm ?

**Comment by RoomToAVue** [comment]
HeroDevs can help with that, email steven@herodevs.com
